### DISCUSSION TITLE
Pimcore Academy > Pimcore Developer Senior > Exam for Course 'Caching'

### DISCUSSION THREAD

**Discussion by mabi-travix** [pimcore/pimcore#12074] [Q&A / Support] [answered]
Hi, 
I don't understand question 5 in Exam for Course 'Caching' - all combinations I've tried (I've tried six so far!) seems wrong and now I believe there must be a bug and/or missing information in the academy:  

**Question 5: What information is part of output cache key?**
- [x] Parameters
- [x] URL
- [x] WebP Support
- [ ] Screen resolution

I've tried to find **any** useful information in all of the academy videos, all of the documentation - including symfony documentation. I've watched/read all that so many times and researched all over the internet to find the necessary explanations about the information parts of cache keys but I couldn't find anything helpful to this question.  

I've even tried all different kinds of possible answers I could imagine and am blocked now for another week again which is very frustrating because I just can't find any helpful information about that at all - so I'll have to trial&error multiple weeks now until I **maybe** pass randomly, which shouldn't be the point of an academy...

Please provide better specific informations to cache key in the documentation and/or videos - I don't think that I'm the only one struggling hard on this one. I don't want to trial&error anymore, I need to understand it!  

I also would be very thankful for the correct answer (if there even is one) and explanation why :(

**Comment by kingjia90** [comment, marked as answer]
All except screen resolution, since it would be difficult and counter-productive to cover every screen resolution (and most likely there would be responsive design anyway). 
The WebP Support needs to be checked, this one is tricky, i've also got stucked on it at the time.

**Reply by mabi-travix** [reply]
Thanks for the reply - That's correct.  
But I would very much appreciate better documentation and/or explanation in the video for other people for the future too. :+1: 
IMO the documentation of cache key in general - with WebP, Parameters and URL - is missing completely.

@fashxp would that be possible to double check please?

**Reply by fashxp** [reply]
actually this is explained explicitly in the first video here https://pimcore.com/academy/course/Caching~13508/chapter-13512 ... last 20 seconds...

**Reply by mabi-travix** [reply]
Honestly I've already had parts of the information from the video, but imo that's not enough explanation to answer the question as a learning beginner.  

I understand URL is part of cacheKey and somehow WebP too (I've overheared the WebP info everytime for example so ok that's clearly my fault), but I personally wouldn't get that "Parameters" are a thing too.  

Maybe all the informations are given, but I think it's not clear enough for everyone (like me).  
At least a few words in the documentation to clarify those things would be great for beginners that aren't sure about that question.

**Reply by mzeitlho** [reply]
sorry to mention that, but even to me this information was pretty unclear. ^^  I would really recommend to explain that a little more in detail in the documentation. I figured out about url and params due to a code snipplet in the documentation of the cache in general, where $cacheKey ist set to an md5-encrypted uri, but the only source of the additional information about WebP i found is literally the last 20 seconds of the video - where it rather sounds like an overal additional information.
